M1FExplained/Chapter25/Exercises/Exercise01.lean

```diff
@@ -0,0 +1,197 @@
+import Mathlib.Algebra.Group.Defs
+import Mathlib.Data.Complex.Basic
+import Mathlib.Analysis.Complex.Basic
+import Mathlib.Data.Nat.Basic
+abbrev complex_size_1 :=
+{ z : ℂ // ‖z‖  = 1 }
+
+instance one : One complex_size_1 := ⟨1,  by exact CstarRing.norm_one⟩
+
+@[simp]
+lemma coe_one : (↑(1 : complex_size_1) : ℂ) = 1 := by rfl
+
+lemma mul_closed (a b : complex_size_1) : ‖(a : ℂ) * b‖ = 1 := by
+  rw [norm_mul, a.2,b.2,mul_one]
+
+--- part i
+
+instance : Mul complex_size_1 where mul a b:= ⟨a.1 * b.1, mul_closed a b⟩
+
+@[simp]
+lemma coe_mul (a b : complex_size_1) : (↑(a * b) : ℂ) = ↑a * ↑b := by
+  rfl
+
+lemma ne_zero (a : complex_size_1) : (a : ℂ) ≠ 0 := by
+  intro h
+  apply_fun norm at h
+  rw [a.2, norm_zero,] at h
+  norm_num at h
+
+noncomputable instance : Group complex_size_1 where
+  mul a b:= ⟨a.1 * b.1, mul_closed a b⟩
+  mul_assoc := by 
+    intro a b c
```

Comment:
Can you do something likw `mul_assoc a b c := by` or `mul_assoc {a b c} := by` or `mul_assoc {a, b, c} := by -- to avoid the `intro`? I can't remember the syntax but I think something like this works